File: typescript/src/types/services.ts
import { ServiceStatus, Currency, Money, DateTime, UUID } from './common';

/**
 * Service response object
 */
export interface Service {
  id: UUID;
  name: string;
  slug: string;
  description: string | null;
  shortDesc: string | null;
  category: string | null;
  price: Money | null;
  currency: Currency;
  durationMins: number | null;
  imageUrl: string | null;
  status: ServiceStatus;
  featured: boolean;
  sortOrder: number;
  metaJson: any | null;
  createdAt: DateTime;
  updatedAt: DateTime;
}

/**
 * Create service request body
 */
export interface CreateServiceBody {
  projectId: UUID;
  name: string;
  slug?: string;
  description?: string | null;
  shortDesc?: string | null;
  category?: string | null;
  priceCents?: number | null;
  currency?: Currency;
  durationMins?: number | null;
  imageUrl?: string | null;
  status?: ServiceStatus;
  featured?: boolean;
  sortOrder?: number;
  metaJson?: any | null;
}

/**
 * Update service request body
 */
export interface UpdateServiceBody {
  name?: string;
  slug?: string;
  description?: string | null;
  shortDesc?: string | null;
  category?: string | null;
  priceCents?: number | null;
  currency?: Currency;
  durationMins?: number | null;
  imageUrl?: string | null;
  status?: ServiceStatus;
  featured?: boolean;
  sortOrder?: number;
  metaJson?: any | null;
}

/**
 * Service list query parameters
 */
export interface ServiceListParams {
  projectId?: UUID;
  search?: string;
  category?: string;
  status?: ServiceStatus;
  featured?: boolean;
  limit?: number;
  page?: number;
}

/**
 * Service statistics response
 */
export interface ServiceStats {
  total: number;
  active: number;
  inactive: number;
  draft: number;
  featured: number;
}

File: python/sakha_cms/models/services.py
"""Services module models."""

from pydantic import BaseModel, ConfigDict, Field
from datetime import datetime
from typing import Optional


class Service(BaseModel):
    """Service."""

    model_config = ConfigDict(populate_by_name=True)

    id: str
    project_id: Optional[str] = Field(None, alias="project_id")
    name: str
    slug: str
    description: Optional[str] = None
    image_url: Optional[str] = Field(None, alias="image_url")
    price: float
    duration_minutes: Optional[int] = Field(None, alias="duration_minutes")
    category: Optional[str] = None
    is_active: bool = Field(default=True, alias="is_active")
    is_featured: bool = Field(default=False, alias="is_featured")
    sort_order: int = Field(default=0, alias="sort_order")
    created_at: datetime = Field(alias="created_at")
    updated_at: datetime = Field(alias="updated_at")


class CreateServiceBody(BaseModel):
    """Create service request."""

    model_config = ConfigDict(populate_by_name=True)

    name: str
    slug: Optional[str] = None
    description: Optional[str] = None
    image_url: Optional[str] = Field(None, alias="imageUrl")
    price: float
    duration_minutes: Optional[int] = Field(None, alias="durationMinutes")
    category: Optional[str] = None
    is_active: bool = Field(default=True, alias="isActive")
    is_featured: bool = Field(default=False, alias="isFeatured")
    sort_order: int = Field(default=0, alias="sortOrder")


class UpdateServiceBody(BaseModel):
    """Update service request."""

    model_config = ConfigDict(populate_by_name=True)

    name: Optional[str] = None
    slug: Optional[str] = None
    description: Optional[str] = None
    image_url: Optional[str] = Field(None, alias="imageUrl")
    price: Optional[float] = None
    duration_minutes: Optional[int] = Field(None, alias="durationMinutes")
    category: Optional[str] = None
    is_active: Optional[bool] = Field(None, alias="isActive")
    is_featured: Optional[bool] = Field(None, alias="isFeatured")
    sort_order: Optional[int] = Field(None, alias="sortOrder")


class ServiceStats(BaseModel):
    """Service statistics."""

    model_config = ConfigDict(populate_by_name=True)

    total_services: int = Field(alias="totalServices")
    active_services: int = Field(alias="activeServices")
    featured_services: int = Field(alias="featuredServices")
    total_revenue: float = Field(alias="totalRevenue")
    average_price: float = Field(alias="averagePrice")
